### PR TITLE
Fixes #2402. Allowing some keywords in namespaces

### LINKED ISSUE #2402: Reserved words cannot be part of the namespace
Currently reserved words like "status" or "fault" cannot be part of a namespace name.
It needs to be decided whether to allow the usage or it must be prominently documented that this is not permitted (similar to #2382). 
Error messages must be displayed in case the user enters an invalid namespace name (see also #2394).

### PR DESCRIPTION
**Pull request by kolotu**
Signed-off-by: Kevin Olotu <kevin.olotu@bosch.io>

### CODE REVIEW

**Review by mena-bosch — COMMENTED** [review]
Small change set looks great, but I would definitely add coverage for this.

**Comment by mena-bosch** [comment]
> Small change set looks great, but I would definitely add coverage for this.

Afterthought: coverage aside, I would also advise for a detailed list of acceptable vs non-acceptable keywords in the namespace docs.

**Comment by kolotu** [comment]
Added coverage, docs and merged.

File: repository/repository-core/src/main/java/org/eclipse/vorto/repository/services/ServiceValidationUtil.java [hunk 1]
```diff
@@ -42,6 +45,12 @@ public static void validateNamespaceName(String namespaceName) throws NameSyntax
           .format("[%s] is not a valid namespace name - aborting namespace creation.",
               namespaceName));
     }
+
+    if (Arrays.stream(namespaceName.split("\\.")).anyMatch(IllegalNamespacePartials.PARTIALS::contains)) {
```
[the remark this answers is not in the document]
Comment (mena-bosch) [reply repository/repository-core/src/main/java/org/eclipse/vorto/repository/services/ServiceValidationUtil.java:49]: Unless the namespace has always been lowercased by then, that is...

**Pull request merged**